### PR TITLE
Check PHP Version and Presence of WooCommerce before Intitialization

### PR DESCRIPTION
**Pull request by lucapircher**
## TLDR;

- checks if the plugin is compatible with the currently used PHP version
  - if not compatible an error messag is shown in the backend and the initialization is aborted
  - if compatible the plugin initializes
- adds a phpcs configuration file which extends PSR12 and the WordPress Docs Style
  - updates the code to match this style

### CODE REVIEW

File: .phpcs.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    <rule ref="PSR12">
+        <exclude name="PSR12.Files.FileHeader" />
+    </rule>
+    <rule ref="vendor/wp-coding-standards/wpcs/WordPress-Docs/ruleset.xml" />
+
+    <rule ref="PSR1.Methods.CamelCapsMethodName">
+        <exclude-pattern>/tests/*.php</exclude-pattern>
+    </rule>
+
```
Comment (philipp-hemmer) [inline .phpcs.xml:27]: too many breaks

File: includes/GbWeiss.php [hunk 2]
```diff
@@ -23,19 +28,100 @@ final class GbWeiss
...
+
+        return true;
+    }
+
+    /**
+     * Checks if the plugin is compatible with the current PHP version.
+     */
+    private static function pluginIsCompatibleWithCurrentPhpVersion(): bool
+    {
+        if (PHP_MAJOR_VERSION === 8) {
```
Comment (philipp-hemmer) [inline includes/GbWeiss.php:83]: i'd maybe shorten it to `return (PHP_MAJOR_VERSION === 8) || (PHP_MAJOR_VERSION === 7 && PHP_MINOR_VERSION >= 2)`

File: includes/GbWeiss.php [hunk 3]
```diff
@@ -23,19 +28,100 @@ final class GbWeiss
...
+     * Shows the passed message as an error in the admin panel
+     *
+     * @param string $message The message to be shown in the admin panel.
+     */
+    private static function showWordpressAdminErrorMessage(string $message): void
+    {
+        add_action(
+            "admin_notices",
+            function () use ($message) {
+                ?>
```
Comment (Martin-Welte) [inline includes/GbWeiss.php:119]: we could refactor this later. i include twig with the options page

**Review by Martin-Welte — APPROVED** [review]

File: includes/GbWeiss.php (hunk 2 above)
Comment (lucapircher) [reply includes/GbWeiss.php:83]: That would result in  a line longer than 80 characters. That's the reason why I've split it into two if statements.

**Review by philipp-hemmer — APPROVED** [review]

File: includes/GbWeiss.php (hunk 3 above)
Comment (lucapircher) [reply includes/GbWeiss.php:119]: Ah nice. We should do that later on.

**Pull request merged**